app/views/comments/_comment.html.erb

```diff
@@ -10,13 +10,15 @@
       <p><%= comment.body %></p>
     </div>
     <div>
-      <%= link_to edit_comment_path(comment), class: "btn btn-link btn-sm text-muted" do %>
-        <i class="fas fa-edit fa-fw"></i>
-      <% end %>
+      <% if comment.author == current_user %>
```

Comment:
Notice here that you have some logic to show or not to some users? This is a clue that it should be using authorization/policies. You can replace this by `if policy(comment).edit?`, this way all the logic stays within the policy file, and here we are calling the policy to check whether a user can see this or not.

Reply:
I will definitely update this. Thank you